privacyidea/lib/resolvers/LDAPIdResolver.py

```diff
@@ -665,18 +719,45 @@ def _attributes_to_ldap_attributes(self, attributes):
         for fieldname, value in attributes.iteritems():
             if self.map.get(fieldname):
                 if fieldname == "password":
-                    password = value 
-                    # Create a {SSHA} password
-                    salt = geturandom(4)
-                    hr = hashlib.sha1(password)
-                    hr.update(salt)
-                    ldap_attributes[self.map.get(fieldname)] = \
-                        "{SSHA}" + hr.digest() + salt
+                    # Variable value may be either a string or a list
+                    # so catch the TypeError exception if we get the wrong
+                    # variable type
+                    try:
+                        pw_hash = self._create_ssha(value[1][0])
+                        value[1][0] = pw_hash
+                        ldap_attributes[self.map.get(fieldname)] = value
+                    except TypeError as e:
+                        pw_hash = self._create_ssha(value)
+                        ldap_attributes[self.map.get(fieldname)] = pw_hash
                 else:
                     ldap_attributes[self.map.get(fieldname)] = value 
 
         return ldap_attributes
 
+    @staticmethod
+    def _create_ssha(password):
```

Comment:
I can take a look at this later. There are also several password hash thingys in the sql resolver.